File: 0547-number-of-provinces/0547-number-of-provinces.py
class Solution:
    def findCircleNum(self, isconnected: List[List[int]]) -> int:
        vis=[0]*len(isconnected)
        count=0
        for i in range(len(vis)):
            if vis[i]==0:
                count+=1
                q=[]
                q.append(i)
                while len(q)!=0:
                    val=q.pop()
                    for j in range(len(isconnected)):
                        if isconnected[val][j]==1 and vis[j]==0:
                            vis[j]=1
                            q.append(j)
        return count


        

File: 0547-number-of-provinces/0547-number-of-provinces.cpp
class Solution {
public:
    int findCircleNum(vector<vector<int>>& isconnected) {
        vector<int>vis(isconnected.size(),0);
        int count=0;
        for(int i=0;i<vis.size();i++){
            if(vis[i]==0){
                queue<int>q;
                q.push(i);
                count++;
                while(!q.empty()){
                    int val=q.front();
                    q.pop();
                    for(int it=0;it<isconnected[val].size();it++){
                        if(isconnected[val][it]==1 && vis[it]==0){
                            q.push(it);
                            vis[it]=1;
                        }
                    }

                }

            }
        }
        return count;
    }
};